nixos/modules/services/monitoring/vnstat.nix

```diff
@@ -0,0 +1,21 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.vnstat;
+in {
+  options.services.vnstat = {
+    enable = mkEnableOption "If enabled, vnstatd will be started to update the network usage statistics.";
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.vnstat = {
+      description = "vnStat network traffic monitor";
+      path = [ pkgs.coreutils ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig.ExecStart = "${pkgs.vnstat}/bin/vnstatd -n";
+      serviceConfig.ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
```

Comment:
@joachifm can you elaborate on this, how would you do this? Is there documentation or examples you can point me to? I have seen that vnstatd supports being run as a different user, maybe this simplifies things.